src/circulation.cpp

```diff
@@ -0,0 +1,116 @@
+// TODO: __example
+
+#include <npe.h>
+#include <typedefs.h>
+
+
+
+
+
+
+#include <igl/circulation.h>
+
+const char* ds_circulation = R"igl_Qu8mg5v7(
+
+Return list of faces around the end point of an edge. Assumes
+  data-structures are built from an edge-manifold **closed** mesh.
+
+Parameters
+----------
+e  index into E of edge to circulate
+ccw  whether to _continue_ in ccw direction of edge (circulate around
+  E(e,1))
+EMAP #F*3 list of indices into E, mapping each directed edge to unique
+  unique edge in E
+EF  #E by 2 list of edge flaps, EF(e,0)=f means e=(i-->j) is the edge of
+  F(f,:) opposite the vth corner, where EI(e,0)=v. Similarly EF(e,1) "
+  e=(j->i)
+EI  #E by 2 list of edge flap corners (see above).
+
+Returns
+-------
+Returns list of faces touched by circulation (in cyclically order).
+
+See also
+--------
+
+Notes
+-----
+None
+
+Examples
+--------
+     
+)igl_Qu8mg5v7";
+
+npe_function(circulation)
+npe_doc(ds_circulation)
+
+npe_arg(e, int)
+npe_arg(ccw, bool)
+npe_arg(emap, dense_int, dense_long)
+npe_arg(ef, npe_matches(emap))
+npe_arg(ei, npe_matches(emap))
+
+
+npe_begin_code()
+
+  // TODO: remove __copy
+  Eigen::VectorXi emap_copy = emap.template cast<int>();
+  Eigen::MatrixXi ef_copy = ef.template cast<int>();
+  Eigen::MatrixXi ei_copy = ei.template cast<int>();
+  igl::circulation(e, ccw, emap_copy, ef_copy, ei_copy);
+  return ;
```

Comment:
`return igl::circulation(e, ccw, emap_copy, ef_copy, ei_copy);`

and drop other